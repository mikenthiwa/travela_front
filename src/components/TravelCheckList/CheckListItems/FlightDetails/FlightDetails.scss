.ticketContainer{
  width: 100%;
  &:last-child{
    margin-bottom: 1em;
  }
  .flight-ticket-details{
    margin-left: 51px;
  }
  .flight-guide{
    float: right;
    color: #3359db;
    padding: 0 2%;
    font-size: 13px;
  }
  .trip-number{
    display: flex;
    justify-content: center;
    border-radius: 50%;
    border: 1px solid #e4e4e4;
    width: 22px;
    height: 22px;
    .counter{
      font-size: 13px;
    }
  }
  .main-ticket-heading{
    padding: 0 18px;
    display: flex;
    flex-direction: row;
    align-items: baseline;
    color: #3359db;

  }
  .form-title{
    padding-left: 9px;
    font-size: 13px;
  }
  .file-upload{
    display: inline-flex;
    justify-content: center;
    align-items: center;
    margin-top: 0.5em;
  }
  hr{
    display: block;
    height: 1px;
    border: 0;
    border-top: 1px solid #e4e4e4;
    margin: 1em 0;
    padding: 0;
  }
  .ticketForm{
    display: flex;
    flex-direction: row;
    width: 76%;
    flex-wrap: wrap;
    padding-top: 10px;
      .ticketInput{
        flex-basis: 40%;
        input{
        line-height: 18px;
        color: #757171!important;

        border-radius: 2px;
        margin: 5px 10px 5px 0;
        background-color: #ffffff;
        border: 1px solid #d4d4d4;
        width: 280px!important;
        height: 32px;
        font-size: 13px;
        }
        input:focus{
          outline: none;
          box-shadow: 0px 0px 3.5px rgba(51, 89, 219, 0.6);
          border: 1px solid rgba(51, 89, 219, 0.7);
        }
        label{
          color: #999999;
          font-size: 13px;
        }
        .react-datepicker.calendar-body {
          box-shadow: 0px 1px 3px 1px rgba(0,0,0,.2);
        }
        }
        .return-form{
          display: flex;
          flex-direction: row;
          width: 100%;
          flex-wrap: wrap;
        }
        .error-div{
          width: 100%;
          font-size: 12px;
          margin-top: -9px;
          margin-bottom: 4px;
        }
        .error{
          color: #ff0000;
        }

    }

    .calendar-icon{
      display: none;
    }

    form .form-input {
      margin: 0!important;
    }

    @media (max-width: 450px){
      form .form-input .date-wrapper input, form .form-input .date-wrapper > div input {
        width: 190px!important;
      }
    }
}
