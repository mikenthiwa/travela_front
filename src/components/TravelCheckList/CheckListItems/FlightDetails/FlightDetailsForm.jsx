import React, { Component, Fragment } from 'react';
import { PropTypes } from 'prop-types';
import moment from 'moment';
import _ from 'lodash';
import DateInput from '../../../Forms/FormsAPI/Input/InputFields/Date';
import FileUploadField from '../FileUploadField/index';

class FlightDetailsForm extends Component {
  state = {
    departureTime: '',
    arrivalTime: '',
    flightNumber: '',
    airline: '',
    error: {travelError: '', returnError: ''},
    returnDepartureTime: '',
    returnArrivalTime: '',
    returnAirline: '',
    returnFlightNumber: ''
  };

  componentDidMount = () => {
    const {
      checklist: {
        ticketDetails: {
          submissions: [item]
        }
      }
    } = this.props;
    if (item) {
      const { departureTime, arrivalTime, flightNumber, airline,returnDepartureTime,
        returnArrivalTime, returnAirline, returnFlightNumber
      } = item.userUpload;
      this.setState({
        departureTime, arrivalTime, flightNumber, airline, returnDepartureTime,
        returnArrivalTime, returnAirline, returnFlightNumber
      });
    }
  };

  validateDates = (arrivalTime, departureTime, errorType) => {
    let isValid = (arrivalTime && departureTime && arrivalTime > departureTime);
    this.setState(prevState => ({
      error: {
        ...prevState.error,
        [errorType]: isValid ? '' : 'Arrival Time should be greater than Departure time'
      }
    }));
    return isValid;
  };

  validateDate = () => {
    const { tripType } = this.props;
    const {departureTime, arrivalTime, returnDepartureTime, returnArrivalTime} = this.state;
    let result = this.validateDates(arrivalTime, departureTime, 'travelError');
    if( tripType === 'return' && (returnArrivalTime && returnDepartureTime)){
      result &= this.validateDates(returnArrivalTime, returnDepartureTime, 'returnError');
    }
    return result;
  };

  submitFlightDetails = () => {
    const {
      departureTime, arrivalTime, airline, flightNumber, returnDepartureTime,
      returnArrivalTime, returnAirline, returnFlightNumber
    } = this.state;
    const {tripId, checklist: { ticketDetails: { id }}, request, postSubmission
    } = this.props;
    const checkId = `${tripId}-${id}`;
    const file = _.pickBy({ departureTime, arrivalTime, airline, flightNumber, returnDepartureTime,
      returnArrivalTime, returnAirline, returnFlightNumber}, _.identity);  
    const formData = { tripId, file };
    postSubmission({ formData, checklistItemId: id }, checkId, request.id);
  };

  handleFlightDetailsSubmit = () => {
    this.validateDate() && this.submitFlightDetails();
  };

  handleDateInputChange = (date, name) => {
    this.setState({
      [name]: this.formatDateTime(date)
    }, this.validateDate);
  };

  handleTextInputChange = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };

  formatDateTime = date => moment(date).format('YYYY-MM-DDTHH:mm');

  renderDateTimeInput = (label, name, tripId, value, min, max) => {
    return (
      <div className="ticketInput form-input">
        <label htmlFor={name}>{label}</label>
        <DateInput
          onBlur={this.handleFlightDetailsSubmit}
          onChange={date => this.handleDateInputChange(date, name)}
          onChangeRaw={date => this.handleDateInputChange(date, name)}
          showTimeSelect
          name={name}
          minimumDate={min !== 'Invalid date' ? moment(min) : null}
          maximumDate={max !== 'Invalid date' ? moment(max) : null}
          dateFormat="MMMM DD, YYYY hh:mm A"
          timeFormat="hh:mm A"
          value={value ? moment(value) : null}
        />
      </div>
    );
  };

  renderTicketInput = (type, name, label, value) => {
    return (
      <div className="ticketInput form-input">
        <label htmlFor={name}>{label}</label>
        <input
          type={type}
          name={name}
          placeholder={label}
          className={name}
          value={value}
          onChange={this.handleTextInputChange}
          onBlur={this.handleFlightDetailsSubmit}
          required
        />
      </div>
    );
  };

  renderIsUploading = () => {
    const {
      fileUploads: { isUploading },
      checkId
    } = this.props;
    return (
      <div className="loader-file">
        {isUploading && isUploading.match(checkId) && (
          <div className="submission-progresbar">
            <div className="submission-progress__spinner" />
            <div id="submission-progress" className="submission-progress__">
              Submitting....
            </div>
          </div>
        )}
      </div>
    );
  };

  renderReturnDetailsFields = () => {
    const { 
      returnDepartureTime, returnArrivalTime, returnFlightNumber,
      returnAirline ,error
    } = this.state;
    const { tripId, trip } = this.props;
    return(
      <div className="return-form">
        {this.renderDateTimeInput(
          'Return Departure Time', 'returnDepartureTime',
          tripId, returnDepartureTime, this.formatDateTime(trip.returnDate),
          this.formatDateTime(trip.returnDate))}
        {this.renderDateTimeInput(
          'Return Arrival Time', 'returnArrivalTime',
          tripId, returnArrivalTime, this.formatDateTime(moment(trip.returnDate).add(1, 'hours')),
          this.formatDateTime(moment(trip.returnDate).add(1, 'days')))}
        <div className="error-div"><span className="error">{error.returnError}</span></div>
        {this.renderTicketInput(
          'text','returnFlightNumber','Return Flight Number', returnFlightNumber)}
        {this.renderTicketInput(
          'text', 'returnAirline', 'Return Airline', returnAirline)}
      </div>
    );
    
  }

  renderTicketForm = () => {
    const {
      departureTime,
      arrivalTime,
      flightNumber,
      airline,
      error
    } = this.state;

    const {
      uploadFile, request, downloadAttachments, fileUploads, postSubmission,
      tripId, trip, checklist, counter, checkId, tripType
    } = this.props;
    return (
      <div className="main-div">
        <div className="main-ticket-heading">
          <div className="trip-number">
            <span className="counter">{counter}</span>
          </div>
          <div className="form-title">
            {`Flight ${counter} - `}
            {`${trip.origin.split(', ')[1]} to ${
              trip.destination.split(', ')[1]
            }`}
          </div>
        </div>
        <div className="flight-ticket-details">
          <form className="ticketForm">
            {this.renderDateTimeInput(
              'Departure Time',
              'departureTime',
              tripId,
              departureTime,
              this.formatDateTime(trip.departureDate),
              this.formatDateTime(trip.returnDate)
            )}
            {this.renderDateTimeInput(
              'Arrival Time',
              'arrivalTime',
              tripId,
              arrivalTime,
              this.formatDateTime(trip.departureDate),
              this.formatDateTime(trip.returnDate)
            )}
            <div className="error-div"><span className="error">{error.travelError}</span></div>
            {this.renderTicketInput('text','flightNumber','Flight Number', flightNumber)}
            {this.renderTicketInput('text', 'airline', 'Airline', airline)}
            
            {tripType === 'return' && this.renderReturnDetailsFields()}
          </form>
          
          <div className="file-upload">
            <FileUploadField
              uploadFile={uploadFile}
              checklistItem={checklist['ticket']}
              tripId={tripId}
              requestId={request.id}
              itemType="travelTicket"
              downloadAttachments={downloadAttachments}
              fileUploads={fileUploads}
              checkId={checkId}
              postSubmission={postSubmission}
            />
            {this.renderIsUploading()}
          </div>
        </div>
      </div>
    );
  };

  renderBreakLine = () => {
    const { counter } = this.props;
    return counter > 1 && <hr />;
  };

  render() {
    return (
      <div>
        {this.renderBreakLine()}
        {this.renderTicketForm()}
      </div>
    );
  }
}

FlightDetailsForm.defaultProps = {
  counter: 1
};

FlightDetailsForm.propTypes = {
  tripId: PropTypes.string.isRequired,
  checklist: PropTypes.object.isRequired,
  request: PropTypes.object.isRequired,
  postSubmission: PropTypes.func.isRequired,
  uploadFile: PropTypes.func.isRequired,
  trip: PropTypes.object.isRequired,
  downloadAttachments: PropTypes.func.isRequired,
  fileUploads: PropTypes.func.isRequired,
  counter: PropTypes.number,
  checkId: PropTypes.string.isRequired,
  tripType: PropTypes.string.isRequired
};

export default FlightDetailsForm;
