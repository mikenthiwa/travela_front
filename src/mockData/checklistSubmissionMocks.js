export const tripRequest = {
  createdAt: '2018-12-19T15:32:39.167Z',
  deletedAt: null,
  department: 'Talent & Development',
  gender: 'Male',
  id: 'w05yTx4aV',
  manager: 'Moffat Gitau',
  name: '4fr0c0d3 Gitau',
  picture: 'https://lh3.googleusercontent.com/-GndNwsNM0HI/AAAAAAAAAAI/AAAAAAAAAAc/0kPTNBhCkXU/photo.jpg?sz=50',
  role: 'Software developer',
  status: 'Approved',
  travelCompletion: '100% complete',
  tripType: 'return',
  trips: [
    {
      bedId: 10945,
      beds: [
        {
          bedName: 'Bed 1',
          booked: false,
          createdAt: '2018-08-16T11:11:52.181Z',
          id: 10945,
          roomId: '10946',
          rooms: {
            bedCount: 2,
            createdAt: '2018-08-16T11:11:52.181Z',
            faulty: false,
            guestHouseId: 10948,
            guestHouses: {
              bathRooms: 3,
              createdAt: '2018-08-16T11:11:52.181Z',
              disabled: false,
              houseName: 'Guest House B',
              id: 10948,
              imageUrl: 'guest.com',
              genderPolicy: 'unisex',
              location: 'Nairobi, Kenya',
              updatedAt: '2018-08-16T11:11:52.181Z',
              userId: '-LTMfmwvXO0D9BQ8fXgl',
            },
            id: 10945,
            isDeleted: false,
            roomName: 'Nyati',
            roomType: 'ensuite',
            updatedAt: '2018-08-16T11:11:52.181Z',
          },
          updatedAt: '2018-08-16T11:11:52.181Z',
        },
      ],
      checkInDate: null,
      checkOutDate: null,
      checkStatus: 'Not Checked In',
      createdAt: '2018-12-19T15:32:39.177Z',
      deletedAt: null,
      departureDate: '2019-07-08',
      destination: 'Nairobi, Kenya',
      id: 'ktgbyIEjWb',
      lastNotifyDate: null,
      notificationCount: 0,
      origin: 'Lagos, Nigeria',
      requestId: 'w05yTx4aV',
      returnDate: '2019-07-11',
      travelCompletion: false,
      updatedAt: '2018-12-19T15:32:39.177Z',
    },
  ],
  updatedAt: '2018-12-19T15:32:39.167Z',
  userId: '-LTMfmwvXO0D9BQ8fXgl',
};

export const LagosExtraSubmission =   {
  destinationName: 'Lagos, Nigeria',
  tripId: 'NumC5-pK7G',
  checklist: [
    {
      id: '9',
      name: 'Travel stipend',
      requiresFiles: true,
      destinationName: 'Lagos, Nigeria',
      deleteReason: null,
      resources: [],
      submissions:[]

    }
  ]
};

