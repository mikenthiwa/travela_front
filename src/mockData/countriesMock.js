export const newCountry = {
  countries : ['abifi', 'asee']
};

const countryData = [
  {
      id: 54,
      country: 'abifi',
      createdAt: '2019-05-17T12:13:02.492Z',
      updatedAt: '2019-05-17T12:13:02.492Z',
      regionId: '1002'
  },
  {
      id: 55,
      country: 'asee',
      createdAt: '2019-05-17T12:13:02.492Z',
      updatedAt: '2019-05-17T12:13:02.492Z',
      regionId: '1002'
  }
];

export const createCountryResponse = {
  data: {
    success: true,
    message: 'Country added successfully',
    countries: countryData
  }
};
export const fetchCountriesResponse = {
  data: {
    success: true,
    message: 'Successfully retrieved countries',
    countries: countryData,
    meta: {
      count: 2,
      pageCount: 1,
      currentPage: 1
  }
  }
};

export const regionId = '1002';
export const query ='';
export const error = 'Possible network error, please reload the page';