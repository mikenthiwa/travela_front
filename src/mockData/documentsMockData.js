const documents = [
  {
    id: '1',
    name: 'Passport',
    cloudinary_url: 'https://image.jpg',
    createdAt: '2018-08-16 12:11:52.181+01',
    updatedAt: '2018-08-16 12:11:52.181+01',
  },
  {
    id: '2',
    name: 'Visa',
    cloudinary_url: 'https://image.url',
    createdAt: '2018-08-16 12:11:52.181+01',
    updatedAt: '2018-08-16 12:11:52.181+01',
  },
  {
    id: '3',
    name: 'Yellow Certificate',
    cloudinary_url: 'https://image.doc',
    createdAt: '2018-08-16 12:11:52.181+01',
    updatedAt: '2018-08-16 12:11:52.181+01',
  },
  {
    id: '4',
    name: 'National ID',
    cloudinary_url: 'https://image.url',
    createdAt: '2018-08-16 12:11:52.181+01',
    updatedAt: '2018-08-16 12:11:52.181+01',
  },
  {
    id: '5',
    name: 'Flight Tickets',
    cloudinary_url: 'https://image.png',
    createdAt: '2018-08-16 12:11:52.181+01',
    updatedAt: '2018-08-16 12:11:52.181+01',
  },
];

export default documents;
