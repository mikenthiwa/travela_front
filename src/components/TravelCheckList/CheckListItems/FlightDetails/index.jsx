import React, { Component } from 'react';
import { PropTypes } from 'prop-types';
import FlightDetailsForm from './FlightDetailsForm';
import './FlightDetails.scss';

class FlightDetails extends Component {
  constructor(props){
    super(props);
    let checkID;
  }

  render() {
    const { 
      submissions, request, uploadFile, downloadAttachments, fileUploads,
      postSubmission, tripType 
    } = this.props;
    let counter = 0;
    return (
      <div className="ticketContainer">
        <div className="flight-guide">Flight Application Guide</div>
        <br />
        {submissions.map((submission) => {
          counter++;
          const ticketCheckList = {};
          submission.checklist.map(checklist => {
            if (checklist.name === 'Travel Ticket') {
              ticketCheckList['ticket'] = checklist;
            }
            if (checklist.name === 'Travel Ticket Details') {
              ticketCheckList['ticketDetails'] = checklist;
            }
          });
          const trip = request.trips.find(trip => trip.id === submission.tripId);
          return (
            <FlightDetailsForm
              uploadFile={uploadFile}
              request={request}
              checklist={ticketCheckList}
              tripId={submission.tripId}
              downloadAttachments={downloadAttachments}
              fileUploads={fileUploads}
              postSubmission={postSubmission}
              tripType={tripType}
              handleInputChange={this.handleInputChange}
              trip={trip}
              counter={counter}
              checkId={`${submission.tripId}-${
                ticketCheckList['ticket'].id
              }`}
              key={`${submission.tripId}-${
                ticketCheckList['ticket'].id
              }`}
            />
          ); 
        })}
      </div>
    ); 
  }
}
FlightDetails.defaultProps ={
  downloadAttachments: () => {}
};

FlightDetails.propTypes = {
  request: PropTypes.object.isRequired,
  uploadFile: PropTypes.func.isRequired,
  downloadAttachments: PropTypes.func,
  postSubmission: PropTypes.func.isRequired,
  tripType: PropTypes.string.isRequired,
  fileUploads: PropTypes.func.isRequired,
  submissions: PropTypes.array.isRequired
};

export default FlightDetails;
