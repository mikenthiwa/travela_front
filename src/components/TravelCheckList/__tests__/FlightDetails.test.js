import React from 'react';
import { mount, shallow } from 'enzyme';
import moment from 'moment';
import { tripRequest } from '../../../mockData/checklistSubmissionMocks';
import FlightDetailsForm from '../CheckListItems/FlightDetails/FlightDetailsForm';
import FlightDetails from '../CheckListItems/FlightDetails/index';
import DateInput from '../../Forms/FormsAPI/Input/InputFields/Date';
import { isForInStatement } from 'typescript';

describe('Flight details tab', () => {
  let props = {
    checkId: 'aLvOR5JCk7-2',
    tripId: 'QbSyCm5XIF',
    checklist: {
      ticket: {
        deleteReason: null,
        destinationName: 'Default',
        id: '2',
        name: 'Travel Ticket',
        requiresFiles: true,
        resources: [],
        submissions: [
          {
            checklistItemId: '2',
            checklistSubmissions: { id: '2' },
            createdAt: '2019-06-19T14:33:09.307Z',
            deletedAt: null,
            documentId: null,
            id: 'WdrAZHV-j',
            tripId: 'QDYDpqEE-U',
            updatedAt: '2019-06-19T14:33:09.307Z',
            userResponse: null,
            userUpload: {
              fileName: 'ticket (1) (1).png',
              url:
                'https://res.cloudinary.com/authors-haven/image/upload/v1560954789/zt6ekgifcak8rsxuwzam.png'
            }
          }
        ]
      },
      ticketDetails: {
        deleteReason: null,
        destinationName: 'Default',
        id: '1',
        name: 'Travel Ticket Details',
        requiresFiles: false,
        resources: [
          {
            checklistItemId: '1',
            id: '1',
            label: 'Flight Application Guide',
            link:
              'https://docs.google.com/document/d/17vOCjPE3sgG2OSYV_3ZcpzCg1IbD7dCO8cVa8aBDN_M/edit?usp=drivesdk'
          }
        ],
        submissions: [
          {
            checklistItemId: '1',
            checklistSubmissions: { id: '1' },
            createdAt: '2019-06-19T14:32:42.556Z',
            deletedAt: null,
            documentId: null,
            id: 'kwgrc80D9',
            tripId: 'QDYDpqEE-U',
            updatedAt: '2019-06-20T09:14:38.907Z',
            userResponse: null,
            userUpload: {
              fileName: 'ticket (1) (1).png',
              url:
                'https://res.cloudinary.com/authors-haven/image/upload/v1560954789/zt6ekgifcak8rsxuwzam.png'
            }
          }
        ]
      }
    },
    request: { ...tripRequest },
    postSubmission: jest.fn(),
    tripType: 'return',
    uploadFile: jest.fn(),
    checkId: 'QbSyCm5XIF-1',
    downloadAttachments: jest.fn(),
    fileUploads: jest.fn(),
    trip: {
      accommodationType: 'Hotel Booking',
      bedId: null,
      beds: null,
      checkInDate: null,
      checkOutDate: null,
      checkStatus: 'Not Checked In',
      createdAt: '2019-06-17T14:07:53.243Z',
      deletedAt: null,
      departureDate: '2019-06-17',
      destination: 'Nairobi, Kenya',
      id: 'QDYDpqEE-U',
      lastNotifyDate: null,
      notificationCount: 0,
      origin: 'Kampala, Uganda',
      otherTravelReasons: 'visiting the embassy',
      reasons: null,
      requestId: 'bZzWv8GtE',
      returnDate: '2019-06-18',
      travelCompletion: 'false',
      travelReasons: null,
      updatedAt: '2019-06-17T14:07:53.243Z'
    }
  };

  const wrapper = mount(<FlightDetailsForm {...props} />);
  const instance = wrapper.instance();

  it('hits submitFlightDetails', () => {
    instance.setState({
      departureTime: '2019-07-08T12:00',
      arrivalTime: '2019-07-08T12:00',
      flightNumber: 'KQ 123',
      airline: 'Kenyan airlines',
      error: ''
    });
    instance.handleFlightDetailsSubmit();
  });

  it('submits partial ticket details', () => {
    instance.setState({
      departureTime: '2019-07-08T12:00',
      arrivalTime: '2019-07-08T12:00',
      flightNumber: 'KQ 123',
    });
    instance.handleFlightDetailsSubmit();
  });

  it('tries to submit with arrival time empty', () => {
    instance.setState({
      departureTime: '2019-07-08T12:00',
      arrivalTime: '',
      flightNumber: 'KQ 123',
    });
    instance.handleFlightDetailsSubmit();
  });

  it('tries to submit with arrival time less or eqaul empty', () => {
    instance.setState({
      departureTime: '2019-07-08T12:00',
      arrivalTime: '2019-07-08T12:00',
      flightNumber: 'KQ 123',
      returnArrivalTime: '2019-07-11T12:00',
      returnDepartureTime: '2019-07-11T12:00'
    });
    instance.handleFlightDetailsSubmit();
  });
  it('tries to submit with return arrival time less than return departure time', () => {
    instance.setState({
      departureTime: '2019-07-08T12:00',
      arrivalTime: '2019-07-08T2:00',
      flightNumber: 'KQ 123',
      returnArrivalTime: '2019-07-11T12:00',
      returnDepartureTime: '2019-07-11T12:00'
    });
    instance.handleFlightDetailsSubmit();
  });
  it('empty error keys', () => {
    instance.setState({
      departureTime: '2019-07-08T12:00',
      arrivalTime: '2019-07-08T2:00',
      flightNumber: 'KQ 123',
      returnArrivalTime: '2019-07-11T2:00',
      returnDepartureTime: '2019-07-11T12:00'
    });
    instance.handleFlightDetailsSubmit();
  });

  it('hits handleInputTextChange', () => {
    const event = {
      target: {
        name: '',
        value: ''
      }
    };
    instance.handleTextInputChange(event);
  });

  it('should call the handleDateInputChange', () => {
    const input = wrapper.find('DateInput[name="departureTime"]');
    input.value= moment('2019-07-08T12:00');
    const event = {
      target: {
        value: '2019-07-08T12:00',
        name: 'departureTime',
        type: 'datetime-local'
      }
    };
    instance.handleDateInputChange(moment('2019-07-08T12:00'), event);
  });

  it('should display a react date picker with option to select time', () => {
    wrapper.find('input[name="departureTime"]').simulate('click');
    const datePicker = wrapper.find('DateInput[name="departureTime"]');
    const timePicker = datePicker.find('Time');
    timePicker.at(0).prop('onChange')(moment('2019-07-08T12:00'));
    expect(timePicker.length).toEqual(1);
  });

  it('should hit onChangeRaw', () => {
    wrapper.find('input[name="departureTime"]').simulate('change', {target: {value: 'June 28, 2019 04:00 PM' }});
  });

});

describe('Flight Details', () => {
  let flightProps = {
    submissions: [
      {
        destinationName: 'Kenya',
        checklist: [
          {
            id: '2',
            name: 'Travel Ticket',
            requiresFiles: true,
            destinationName: 'Default',
            deleteReason: null,
            resources: [],
            submissions: [
              {
                id: 'WdrAZHV-j',
                userResponse: null,
                userUpload: {
                  url:
                    'https://res.cloudinary.com/authors-haven/image/upload/v1560954789/zt6ekgifcak8rsxuwzam.png',
                  fileName: 'ticket (1) (1).png'
                },
                tripId: 'QDYDpqEE-U',
                checklistItemId: '2',
                documentId: null,
                createdAt: '2019-06-19T14:33:09.307Z',
                updatedAt: '2019-06-19T14:33:09.307Z',
                deletedAt: null,
                checklistSubmissions: {
                  id: '2'
                }
              }
            ]
          },
          {
            id: '1',
            name: 'Travel Ticket Details',
            requiresFiles: false,
            destinationName: 'Default',
            deleteReason: null,
            resources: [
              {
                id: '1',
                label: 'Flight Application Guide',
                link:
                  'https://docs.google.com/document/d/17vOCjPE3sgG2OSYV_3ZcpzCg1IbD7dCO8cVa8aBDN_M/edit?usp=drivesdk',
                checklistItemId: '1'
              }
            ],
            submissions: [
              {
                id: 'kwgrc80D9',
                userResponse: null,
                userUpload: {},
                tripId: 'QDYDpqEE-U',
                checklistItemId: '1',
                documentId: null,
                createdAt: '2019-06-19T14:32:42.556Z',
                updatedAt: '2019-06-20T12:23:17.980Z',
                deletedAt: null,
                checklistSubmissions: {
                  id: '1'
                }
              }
            ]
          }
        ],
        tripId: 'QDYDpqEE-U',
        tripLocation: 'Nairobi, Kenya',
        tripOrigin: 'Uganda'
      }
    ],
    request: { ...tripRequest },
    uploadFile: jest.fn(),
    downloadAttachments: jest.fn(),
    fileUploads: jest.fn(),
    postSubmission: jest.fn(),
    tripType: 'return',
    trip: {
      accommodationType: 'Hotel Booking',
      bedId: null,
      beds: null,
      checkInDate: null,
      checkOutDate: null,
      checkStatus: 'Not Checked In',
      createdAt: '2019-06-17T14:07:53.243Z',
      deletedAt: null,
      departureDate: '2019-06-17',
      destination: 'Nairobi, Kenya',
      id: 'QDYDpqEE-U',
      lastNotifyDate: null,
      notificationCount: 0,
      origin: 'Kampala, Uganda',
      otherTravelReasons: 'visiting the embassy',
      reasons: null,
      requestId: 'bZzWv8GtE',
      returnDate: '2019-06-18',
      travelCompletion: 'false',
      travelReasons: null,
      updatedAt: '2019-06-17T14:07:53.243Z'
    }
  };

  it('matches snapshot', () => {
    const indexTree = shallow(<FlightDetails {...flightProps} />);
    expect(indexTree).toMatchSnapshot();
  });
});
