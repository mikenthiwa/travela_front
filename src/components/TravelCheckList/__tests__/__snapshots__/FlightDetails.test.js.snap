// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flight Details matches snapshot 1`] = `
<div
  className="ticketContainer"
>
  <div
    className="flight-guide"
  >
    Flight Application Guide
  </div>
  <br />
  <FlightDetailsForm
    checkId="QDYDpqEE-U-2"
    checklist={
      Object {
        "ticket": Object {
          "deleteReason": null,
          "destinationName": "Default",
          "id": "2",
          "name": "Travel Ticket",
          "requiresFiles": true,
          "resources": Array [],
          "submissions": Array [
            Object {
              "checklistItemId": "2",
              "checklistSubmissions": Object {
                "id": "2",
              },
              "createdAt": "2019-06-19T14:33:09.307Z",
              "deletedAt": null,
              "documentId": null,
              "id": "WdrAZHV-j",
              "tripId": "QDYDpqEE-U",
              "updatedAt": "2019-06-19T14:33:09.307Z",
              "userResponse": null,
              "userUpload": Object {
                "fileName": "ticket (1) (1).png",
                "url": "https://res.cloudinary.com/authors-haven/image/upload/v1560954789/zt6ekgifcak8rsxuwzam.png",
              },
            },
          ],
        },
        "ticketDetails": Object {
          "deleteReason": null,
          "destinationName": "Default",
          "id": "1",
          "name": "Travel Ticket Details",
          "requiresFiles": false,
          "resources": Array [
            Object {
              "checklistItemId": "1",
              "id": "1",
              "label": "Flight Application Guide",
              "link": "https://docs.google.com/document/d/17vOCjPE3sgG2OSYV_3ZcpzCg1IbD7dCO8cVa8aBDN_M/edit?usp=drivesdk",
            },
          ],
          "submissions": Array [
            Object {
              "checklistItemId": "1",
              "checklistSubmissions": Object {
                "id": "1",
              },
              "createdAt": "2019-06-19T14:32:42.556Z",
              "deletedAt": null,
              "documentId": null,
              "id": "kwgrc80D9",
              "tripId": "QDYDpqEE-U",
              "updatedAt": "2019-06-20T12:23:17.980Z",
              "userResponse": null,
              "userUpload": Object {},
            },
          ],
        },
      }
    }
    counter={1}
    downloadAttachments={[Function]}
    fileUploads={[Function]}
    key="QDYDpqEE-U-2"
    postSubmission={[Function]}
    request={
      Object {
        "createdAt": "2018-12-19T15:32:39.167Z",
        "deletedAt": null,
        "department": "Talent & Development",
        "gender": "Male",
        "id": "w05yTx4aV",
        "manager": "Moffat Gitau",
        "name": "4fr0c0d3 Gitau",
        "picture": "https://lh3.googleusercontent.com/-GndNwsNM0HI/AAAAAAAAAAI/AAAAAAAAAAc/0kPTNBhCkXU/photo.jpg?sz=50",
        "role": "Software developer",
        "status": "Approved",
        "travelCompletion": "100% complete",
        "tripType": "return",
        "trips": Array [
          Object {
            "bedId": 10945,
            "beds": Array [
              Object {
                "bedName": "Bed 1",
                "booked": false,
                "createdAt": "2018-08-16T11:11:52.181Z",
                "id": 10945,
                "roomId": "10946",
                "rooms": Object {
                  "bedCount": 2,
                  "createdAt": "2018-08-16T11:11:52.181Z",
                  "faulty": false,
                  "guestHouseId": 10948,
                  "guestHouses": Object {
                    "bathRooms": 3,
                    "createdAt": "2018-08-16T11:11:52.181Z",
                    "disabled": false,
                    "genderPolicy": "unisex",
                    "houseName": "Guest House B",
                    "id": 10948,
                    "imageUrl": "guest.com",
                    "location": "Nairobi, Kenya",
                    "updatedAt": "2018-08-16T11:11:52.181Z",
                    "userId": "-LTMfmwvXO0D9BQ8fXgl",
                  },
                  "id": 10945,
                  "isDeleted": false,
                  "roomName": "Nyati",
                  "roomType": "ensuite",
                  "updatedAt": "2018-08-16T11:11:52.181Z",
                },
                "updatedAt": "2018-08-16T11:11:52.181Z",
              },
            ],
            "checkInDate": null,
            "checkOutDate": null,
            "checkStatus": "Not Checked In",
            "createdAt": "2018-12-19T15:32:39.177Z",
            "deletedAt": null,
            "departureDate": "2019-07-08",
            "destination": "Nairobi, Kenya",
            "id": "ktgbyIEjWb",
            "lastNotifyDate": null,
            "notificationCount": 0,
            "origin": "Lagos, Nigeria",
            "requestId": "w05yTx4aV",
            "returnDate": "2019-07-11",
            "travelCompletion": false,
            "updatedAt": "2018-12-19T15:32:39.177Z",
          },
        ],
        "updatedAt": "2018-12-19T15:32:39.167Z",
        "userId": "-LTMfmwvXO0D9BQ8fXgl",
      }
    }
    tripId="QDYDpqEE-U"
    tripType="return"
    uploadFile={[Function]}
  />
</div>
`;
